pkg/memory-monitor/src/monitor/memory-monitor-handler.sh

```diff
@@ -117,16 +120,26 @@ show_pid_mem_usage() {
       # Process is gone
       continue
     fi
-    name=$(ps -p "$pid" -o cmd=)
+    if ! name=$(ps -p "$pid" -o cmd=) ; then
+      # Process is gone
+      continue
+    fi
     if [ "$detailed" -eq 1 ]; then
       echo "Process $name PID: $pid" >> "$output_file"
     fi
     # Get the memory usage according to ps
-    ps_rss=$(ps -p "$pid" -o rss= | tr -d ' ')
+    if ! ps_rss=$(ps -p "$pid" -o rss= | tr -d ' ') ; then
+      # Process is gone
```

Comment:
If the process is gone here but existed on line 123 you have already put information about it in the output_file.
Does that matter? If it does then all the output should be done after all of the "gone" checks.

Reply:
The issue is that we can’t find a stable moment when processes exist reliably to capture all data in one go. Processes often spawn and die within the cgroups, which is expected. Therefore, I think it's fine to record processes at the moment we first dump the list but exclude them from detailed analysis if they disappear later.

From my observations, "fat" processes (those with significant resource consumption) tend to stick around, so we shouldn’t lose much critical information this way. On the other hand, if we don’t handle these disappearing processes this way, the script will exit entirely (since we run with set -e), causing us to lose the rest of the statistics. That would be a bigger issue.

To answer your next question in advance, we run the script with -e to make debugging easier. If a critical error occurs, we want it to appear in the handler log's final lines, which are sent to the syslog (only the last 10 lines are included, not the entire log).

Reply:
OK, as long as the analysis can deal with the partial info when the process dies in between these steps.

Reply:
I can add printing to output that the process does not exist anymore. As the analysis of this file is not automated (the output file is the file that is expected to be reviewed by us), it should be enough.

Reply:
Done